app/models/solr_document.rb

```diff
@@ -28,6 +28,18 @@ class SolrDocument
 
   use_extension( Hydra::ContentNegotiation )
 
+  def visibility
+    if rights_override?
+      @visibility ||= VisibilityService.visibility_value(self[Solrizer.solr_name('rights_override')].first)
+    else
+      super
+    end
+  end
+
+  def rights_override?
+    !self[Solrizer.solr_name('rights_override')].nil? && !self[Solrizer.solr_name('rights_override')].first.blank?
```

Comment:
I think we can start making use of some newer syntax available in Hyrax for loading solr properties. In this case the `fetch` option might be nice. I tried to [document this](https://mcritchlow.github.io/hydra/development/2017/05/09/advanced-hydra-camp-day-1.html) from the Hydra Camp (see the Pairing Musical Work section)